cpu/native/syscalls.c

```diff
@@ -422,10 +422,9 @@ int getpid(void)
 int _gettimeofday(struct timeval *tp, void *restrict tzp)
 {
     (void) tzp;
-    timex_t now;
-    xtimer_now_timex(&now);
-    tp->tv_sec = now.seconds;
-    tp->tv_usec = now.microseconds;
+    uint64_t now = xtimer_now_usec64();
+    tp->tv_sec = div_u64_by_1000000(now);
```

Comment:
Is the `div` module available with `native`? Otherwise this needs to be added as a dependency.

Reply:
Why not use regular division on native? Most bigger CPUs will not have the limitations we are trying to avoid with code size and unnecessary runtime calls to software division with constant denominators.

Reply:
First, module `div` is a dependency of xtimer and `_gettimeofday` is only included if MODULE_XTIMER is used. Second, I basically copied from the equivalent function in `newlib_syscalls_`, which used the `div_u64_` division - and yes those optimisation are likely unnecessary for native.

So, shall I change to plain `tp->tv_sec = now / US_PER_SEC`, or leave as is - I don't have a strong opinion here, and also think it doesn't matter that much.

Reply:
Doesn't matter to me which solution is used.

Reply:
I think regardless that due to this line `div` should become a dependency of `native` (though I prefer using `/` to not have that dependency). Not addressing the dependency due to some transitive resolution is a bad idea, if e.g. in the future the `xtimer` dependency is dropped.

Reply:
okay, then I'll go for plain good old `/`.